src/commands/workflow/compile.ts

```diff
@@ -17,9 +18,35 @@ export default class WorkflowCompile extends Command {
 
   async run(): Promise<any> {
     const {args} = this.parse(WorkflowCompile)
-    const definition = await compile.workflow(readFileSync(args.WORKFLOW_FILE))
+    const definition = await compile.workflow(readFileSync(args.WORKFLOW_FILE), async (instanceObject: any) => {
+      if (instanceObject.instanceHash) {
+        return instanceObject.instanceHash
+      }
+      if (instanceObject.service) {
+        return this.serviceToInstance(instanceObject.service)
+      }
+      throw new Error('instance resolution error')
+    })
     this.styledJSON(definition)
     this.spinner.stop()
     return definition
   }
+
+  async serviceToInstance(key: string): Promise<string> {
+    const {services} = await this.api.service.list({})
+    if (!services) throw new Error('no services deployed, please deploy your service first')
+    const match = services.filter(x => x.hash === key || x.sid === key)
+    if (match.length > 1) throw new Error(`multiple services match the following key: ${key}, please use service Hash instead.`)
+    const service = match[0]
+    if (!service.hash) throw new Error('invalid service')
```

Comment:
if service is undefined it throw an error here (i think) because there is not check for `match.length === 0`
```
 ›   Error: Cannot read property 'hash' of undefined
```

Reply:
I get this error by setting a valid `instanceHash` in `service` parameter of a task of the worklow

Reply:
done